main.go

```diff
@@ -2,100 +2,78 @@ package main
 
 import (
 	"fmt"
-	"net"
+	"math"
 	"os/exec"
 	"strconv"
 	"strings"
 	"time"
-)
-
-type progressBar struct {
-	portRange     [2]int
-	current       int
-	percentage    float64
-	lastDisplayed string
-}
-
-func (p *progressBar) setPercentage(i float64) {
-	perc := i / float64((*p).portRange[1]-(*p).portRange[0])
-	(*p).percentage = perc
-}
-
-// Gives go time to close up some of the open connections
-func batchCalls(r [2]int, c chan string, pb *progressBar) {
-	for i := r[0]; i < r[1]; i++ {
-		go pingServerPort(i, c)
-	}
-	j := r[0]
-	for l := range c {
-		j++
-		go func(l string, j int) {
-			if l != "." {
-				fmt.Println(l)
-			}
-		}(l, j)
-		percentageHelper(pb, float64(j-r[0]))
-		if j >= r[1] {
-			return
-		}
-	}
-
-}
-
-func pingServerPort(p int, c chan string) {
-	port := strconv.FormatInt(int64(p), 10)
-	conn, err := net.DialTimeout("tcp", "127.0.0.1:"+port, 5000*time.Millisecond)
 
-	if err == nil {
-		c <- "Port " + port + " is open"
-		conn.Close()
-		return
-	}
-	c <- "."
-	return
-}
+	ap "github.com/drypycode/port-scanner/argparse"
+	ps "github.com/drypycode/port-scanner/portscanner"
+	pb "github.com/drypycode/port-scanner/progressbar"
+	"github.com/sirupsen/logrus"
+)
 
+// GetUlimit ...
 func getUlimit() int {
 	out, err := exec.Command("ulimit", "-n").Output()
 	if err != nil {
 		panic(err)
 	}
-	s := strings.TrimSpace(string(out))
-	i, err := strconv.ParseInt(s, 10, 64)
+	ulimit := strings.TrimSpace(string(out))
+	i, err := strconv.ParseInt(ulimit, 10, 64)
 
 	if err != nil {
 		panic(err)
```

Comment:
IIUC, idiomatic Golang dictates that shit should return (int, err), rather than panic()ing. Calling `panic(err)` is generally reserved for an unrecoverable error. Can the program go on without getting the ulimit?

```golang
func getUlimit() (int, err) {
    ....
}
```